test/extended/util/framework.go

```diff
@@ -390,13 +390,19 @@ func (t *BuildResult) dumpRegistryLogs() {
 	}
 
 	since := time.Now().Sub(*buildStarted)
+
+	// Changing the namespace on the derived client still changes it on the original client
+	// because the kubeFramework field is only copied by reference. Saving the original namespace
+	// here so we can restore it when done with registry logs
+	savedNamespace := t.oc.Namespace()
 	oadm := t.oc.AsAdmin().SetNamespace("default")
 	out, err := oadm.Run("logs").Args("dc/docker-registry", "--since="+since.String()).Output()
 	if err != nil {
 		fmt.Fprintf(g.GinkgoWriter, "Error during log retrieval: %+v\n", err)
 	} else {
 		fmt.Fprintf(g.GinkgoWriter, "%s\n", out)
 	}
+	t.oc.SetNamespace(savedNamespace)
```

Comment:
There really is no validation of the namespace you tell it to use besides setting it inside the kubeFramework struct. AsAdmin() will create a copy of the oc client and point it to the admin.kubeconfig. The problem above is that the copy is only a shallow copy, oadm (which you get by calling oc.AsAdmin()) and oc both point to the same kubeFramework object. It doesn't matter on which you call SetNamespace.